Mastodon/Coordinator/SceneCoordinator.swift

```diff
@@ -529,5 +538,16 @@ private extension SceneCoordinator {
         needs?.context = appContext
         needs?.coordinator = self
     }
-    
+}
+
+//MARK: - MastodonLoginViewControllerDelegate
+
+extension SceneCoordinator: MastodonLoginViewControllerDelegate {
+  func backButtonPressed(_ viewController: MastodonLoginViewController) {
+    viewController.navigationController?.popViewController(animated: true)
+  }
+
+  func nextButtonPressed(_ viewController: MastodonLoginViewController) {
+    viewController.login()
+  }
```

Comment:
Maybe that navigation button event could be handled directly in the view controller.

Reply:
The idea was to _not_ have this in the ViewController itself. A specific ViewController controls a specific view, navigation and flow and how the ViewController is used, is none of its business. Does this make sense?